nuxeo-features/nuxeo-platform-rendition/nuxeo-platform-rendition-core/src/main/java/org/nuxeo/ecm/platform/rendition/service/RenditionServiceImpl.java

```diff
@@ -378,22 +382,30 @@ protected Rendition getRendition(DocumentModel doc, RenditionDefinition renditio
         Rendition rendition;
         boolean isVersionable = doc.isVersionable();
         if (!isVersionable || !doc.isCheckedOut()) {
+            log.debug("Document {} is not versionable nor checked out, trying to find a stored rendition.", doc);
             // stored renditions are only done against a non-versionable doc
             // or a versionable doc that is not checkedout
             rendition = getStoredRenditionManager().findStoredRendition(doc, renditionDefinition);
             if (rendition != null) {
+                log.debug("Found and returning a stored rendition for document {}.", doc);
                 return rendition;
+            } else {
+                log.debug("Found no stored rendition for document {}.", doc);
             }
+        } else {
+            log.debug("Document {} is versionable and checked out, not trying to find any stored rendition.", doc);
         }
 
         rendition = new LiveRendition(doc, renditionDefinition);
 
         if (store) {
             StoredRendition storedRendition = storeRendition(doc, rendition, renditionDefinition);
             if (storedRendition != null) {
+                log.debug("Returning new stored rendition for document {}.", doc);
                 return storedRendition;
             }
         }
+        log.debug("No rendition stored for document {}, returning live rendition.", doc);
```

Comment:
`log.debug("No rendition stored for document {}, returning live rendition.", doc);` will be logged in the two both cases `store is false` and `storedRendition is null` perhaps for more clarity we need to separate these two cases
- when we ask for storing and the returned one is `null`
-no store then we don't need to log or just to say no store asked